lib/Backend/Vault/Kv.hs

```diff
@@ -204,18 +197,17 @@ runVaultIO url token mount = interpret $
                 case statusCode $ responseStatusCode response of
                   404 -> Nothing
                   e -> Just $ OtherError (T.pack $ show e)
-            DecodeFailure text response -> Just MarshallingFailed
-            UnsupportedContentType mediaType response -> Just MarshallingFailed
-            InvalidContentTypeHeader response -> Just MarshallingFailed
-            ConnectionError exception -> Just ConnectError
+            DecodeFailure _ _ -> Just MarshallingFailed
+            UnsupportedContentType _ _ -> Just MarshallingFailed
+            InvalidContentTypeHeader _ -> Just MarshallingFailed
+            ConnectionError _ -> Just ConnectError
 
     -- | Runs an IO action and throws an error if happens.
     embedCatch :: Member (Embed IO) r
                     => Member (Error CofferError) r
-                    => [T.Text]
-                    -> IO a
+                    => IO a
                     -> Sem r a
-    embedCatch path io = embed (catch @ClientError (io <&> Left) (pure . Right . exceptionHandler)) >>=
+    embedCatch io = embed (catch @ClientError (io <&> Left) (pure . Right . exceptionHandler)) >>=
```

Comment:
I suggest renaming to `embedCatchClientError` or to `embedCatch @ClientError` :peka:

Reply:
Done